std/os.zig

```diff
@@ -103,33 +103,45 @@ pub fn getrandom(buf: []u8) GetRandomError!void {
     if (windows.is_the_target) {
         return windows.RtlGenRandom(buf);
     }
-    if (linux.is_the_target) {
-        while (true) {
-            const err = if (std.c.versionCheck(builtin.Version{ .major = 2, .minor = 25, .patch = 0 }).ok) blk: {
-                break :blk errno(std.c.getrandom(buf.ptr, buf.len, 0));
+    if (linux.is_the_target or freebsd.is_the_target) {
+        var buf_slice: []u8 = buf[0..];
+        var total_read: usize = 0;
+        const use_c = if (linux.is_the_target) blk: {
```

Comment:
You don't need to use a block here. Just write:
```zig
const use_c = if (linux.is_the_target) std.c.versionCheck(builtin.Version{ .major = 2, .minor = 25, .patch = 0 }).ok else true;
```
Or even:
```zig
const use_c = (not linux.is_the_target) or std.c.versionCheck(builtin.Version{ .major = 2, .minor = 25, .patch = 0 }).ok;
```

Btw, that version check looks like it's checking a glibc version. What occurs on musl?

Reply:
Sure, I'll change that.

The version check is copied from the previous code. I'm not sure how it behaves on musl.